app/components/post_textbox/post_textbox_base.js

```diff
@@ -106,11 +113,27 @@ export default class PostTextBoxBase extends PureComponent {
             Keyboard.addListener('keyboardDidHide', this.handleAndroidKeyboard);
             BackHandler.addEventListener('hardwareBackPress', this.handleAndroidBack);
         }
+
+        if (this.props.isTimezoneEnabled) {
+            const {data} = await this.props.actions.getChannelTimezones(this.props.channelId);
+            if (data) {
+                this.setState({channelTimezoneCount: data.length}); // eslint-disable-line react/no-did-mount-set-state
+            } else {
+                this.setState({channelTimezoneCount: 0}); // eslint-disable-line react/no-did-mount-set-state
+            }
+        }
     }
 
-    componentWillReceiveProps(nextProps) {
+    async componentWillReceiveProps(nextProps) {
         if (nextProps.channelId !== this.props.channelId || nextProps.rootId !== this.props.rootId) {
-            this.setState({value: nextProps.value});
+            let channelTimezoneCount = 0;
+            if (nextProps.isTimezoneEnabled) {
```

Comment:
You should also check that the previous prop is diff than the nextProp to avoid multiple calls as the nextProp can be true while other props are being updated.

Also as I mention above this can be done in a function.

On top of what I previously mentioned I’m not very keen on adding more code to this lifecycle method as it will make it harder to migrate, remember that this method is now deprecated. Try and think of a different way to achieve this.

Reply:
I did some research and came across [this](https://reactjs.org/docs/react-component.html#componentdidupdate). I have removed all the extra code I added and moved it to the `componenetDidMount` lifecycle method.